--- a/include/package.mk
+++ b/include/package.mk
@@ -69,7 +69,7 @@ $(curdir)/install: $(TMP_DIR)/.build $(curdir)/merge $(if $(CONFIG_TARGET_PER_DE
 	$(file >$(TMP_DIR)/opkg_install_list,\
 	  $(call opkg_package_files,\
 	    $(foreach pkg,$(shell cat $(PACKAGE_INSTALL_FILES) 2>/dev/null),$(pkg)$(call GetABISuffix,$(pkg)))))
-	$(call opkg,$(TARGET_DIR)) install $$(cat $(TMP_DIR)/opkg_install_list)
+	$(call opkg,$(TARGET_DIR)) install --force-overwrite --force-depends $$(cat $(TMP_DIR)/opkg_install_list)
 	@for file in $(PACKAGE_INSTALL_FILES); do \
 		[ -s $$file.flags ] || continue; \
 		for flag in `cat $$file.flags`; do \
@@ -354,3 +354,11 @@ dist:
 
 distcheck:
 	$(Build/DistCheck)
+
+ifndef Package/$(PKG_NAME)/conffiles
+define Package/$(PKG_NAME)/conffiles
+/etc/config/
+/etc/$(PKG_NAME)/
+endef
+endif
+
