--- a/include/image.mk
+++ b/include/image.mk
@@ -537,6 +537,7 @@ define Device/Build/image
 	IMAGE_TYPE=$(word 1,$(subst ., ,$(2))) \
 	IMAGE_FILESYSTEM="$(1)" \
 	IMAGE_PREFIX="$(IMAGE_PREFIX)" \
+	KERNEL_INITRAMFS_IMAGE="$(subst $(IMG_PREFIX_EXTRA),,$(KERNEL_INITRAMFS_IMAGE))" \
 	DEVICE_TITLE="$(DEVICE_TITLE)" \
 	DEVICE_PACKAGES="$(DEVICE_PACKAGES)" \
 	TARGET="$(BOARD)" \

--- a/scripts/json_add_image_info.py
+++ b/scripts/json_add_image_info.py
@@ -48,6 +48,7 @@ def get_titles():
     "profiles": {
         device_id: {
             "image_prefix": getenv("IMAGE_PREFIX"),
+            "image_initramfs": getenv("KERNEL_INITRAMFS_IMAGE"),
             "images": [
                 {
                     "type": getenv("IMAGE_TYPE"),

--- a/scripts/json_overview_image_info.py
+++ b/scripts/json_overview_image_info.py
@@ -47,7 +47,7 @@ def get_initial_output(image_info):
 
 
 if output:
-    default_packages, output["arch_packages"] = run(
+    default_packages, output["arch_packages"], output["kernel_version"] = run(
         [
             "make",
             "--no-print-directory",
@@ -55,6 +55,7 @@ def get_initial_output(image_info):
             "target/linux/",
             "val.DEFAULT_PACKAGES",
             "val.ARCH_PACKAGES",
+            "val.LINUX_VERSION",
         ],
         stdout=PIPE,
         stderr=PIPE,

--- a/package/kernel/mac80211/ath.mk
+++ b/package/kernel/mac80211/ath.mk
@@ -282,7 +282,7 @@ define KernelPackage/ath11k
   TITLE:=Atheros 802.11ax wireless cards support
   URL:=https://wireless.wiki.kernel.org/en/users/drivers/ath11k
   DEPENDS+= @TARGET_ipq60xx +kmod-ath +@DRIVER_11N_SUPPORT +@DRIVER_11AC_SUPPORT \
-	+@DRIVER_11AX_SUPPORT +@DRIVER_11W_SUPPORT +kmod-qca-nss-drv
+	+@DRIVER_11AX_SUPPORT +@DRIVER_11W_SUPPORT +kmod-qca-nss-drv +kmod-hwmon-core
   FILES:= \
 	$(PKG_BUILD_DIR)/drivers/net/wireless/ath/ath11k/ath11k.ko
 endef

--- a/target/linux/ipq60xx/image/Makefile
+++ b/target/linux/ipq60xx/image/Makefile
@@ -104,7 +104,7 @@ define Device/qihoo_v6
 	PAGESIZE := 2048
 	DEVICE_DTS := qcom-ipq6018-qihoo-v6
 	DEVICE_DTS_CONFIG := config@cp03-c1
-	DEVICE_TITLE := Qihoo V6
+	DEVICE_TITLE := Qihoo 360 V6
 	DEVICE_PACKAGES := ath11k-wifi-qihoo-v6 kmod-usb3 kmod-usb-phy-msm
 endef
 TARGET_DEVICES += qihoo_v6

--- a/target/linux/ipq60xx/files-4.4/arch/arm64/boot/dts/qcom/qcom-ipq6018-zn-m2.dts
+++ b/target/linux/ipq60xx/files-4.4/arch/arm64/boot/dts/qcom/qcom-ipq6018-zn-m2.dts
@@ -70,3 +70,15 @@
 		};
 	};
 };
+
+&qusb_phy_0 {
+status = "ok";
+};
+
+&ssphy_0 {
+status = "ok";
+};
+
+&usb3 {
+status = "ok";
+};

--- a/package/firmware/ath11k-wifi/Makefile
+++ b/package/firmware/ath11k-wifi/Makefile
@@ -28,7 +28,7 @@ ALLWIFIBOARDS:= \
   cmiot-ax18 \
 	gl-ax1800 \
 	gl-axt1800 \
-	qihoo_v6
+	qihoo-v6
 
 ALLWIFIPACKAGES:=$(foreach BOARD,$(ALLWIFIBOARDS),ath11k-wifi-$(BOARD))
 
@@ -88,7 +88,7 @@ endef
 $(eval $(call generate-ath11k-wifi-package,cmiot-ax18,CMIOT AX18))
 $(eval $(call generate-ath11k-wifi-package,gl-ax1800,Gl.iNET AX1800))
 $(eval $(call generate-ath11k-wifi-package,gl-axt1800,Gl.iNET AXT1800))
-$(eval $(call generate-ath11k-wifi-package,qihoo_v6,QIHOO 360V6))
+$(eval $(call generate-ath11k-wifi-package,qihoo-v6,QIHOO 360V6))
 
 
 $(foreach PACKAGE,$(ALLWIFIPACKAGES),$(eval $(call BuildPackage,$(PACKAGE))))

--- a/package/feeds/kiddin9/base-files/files/etc/init.d/led
+++ b/package/feeds/kiddin9/base-files/files/etc/init.d/led
@@ -76,6 +76,8 @@ load_led() {
 		"netdev")
 			[ -n "$dev" ] && {
 				echo $dev > /sys/class/leds/${sysfs}/device_name
+				[ -e "/sys/class/leds/${sysfs}/mode" ]  && \
+					echo $mode > /sys/class/leds/${sysfs}/mode
 				for m in $mode; do
 					[ -e "/sys/class/leds/${sysfs}/$m" ] && \
 						echo 1 > /sys/class/leds/${sysfs}/$m

diff --git a/package/firmware/ath11k-wifi/board-qihoo_360v6.bin.IPQ6018 b/package/firmware/ath11k-wifi/board-qihoo-v6.bin.IPQ6018
similarity index 100%
rename from package/firmware/ath11k-wifi/board-qihoo_360v6.bin.IPQ6018
rename to package/firmware/ath11k-wifi/board-qihoo-v6.bin.IPQ6018
